### DISCUSSION TITLE
Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "105a61b6b61f191cb29d62f7e1a2bbd30e4caf737cc32ae0a4606e4f2900c6e4": unable to allocate IP address: Post "http://127.0.0.1:6784/ip/105a61b6b61f191cb29d62f7e1a2bbd30e4caf737cc32ae0a4606e4f2900c6e4": dial tcp 127.0.0.1:6784: connect: connection refused

### DISCUSSION THREAD

**Discussion by vitorafgomes** [coredns/coredns#5147] [Q&A]
Hello, could anyone tell me if this would be a bug or something that I failed to install?

OS: Ubuntu 21.10 aarch64
Host: Raspberry Pi 4 Model B Rev 1.4
Kernel: 5.13.0-1015-raspi
CPU: BCM2835 (4) @ 1.500GHz
Memory: 722MiB / 7809MiB

in the installation I just ran the command kubeadm init

Name: coredns-64897985d-7lcxm
Namespace: kube-system
Priority: 2000000000
Priority Class Name: system-cluster-critical
Node: k8s-w-02/192.168.50.112
Start Time: Sun, 30 Jan 2022 16:20:47 +0000
Labels: k8s-app=kube-dns
pod-template-hash=64897985d
Annotations:
Status: Pending
IP:
IPs:
Controlled By: ReplicaSet/coredns-64897985d
Containers:
coredns:
Container ID:
Image: k8s.gcr.io/coredns/coredns:v1.8.6
Image ID:
Ports: 53/UDP, 53/TCP, 9153/TCP
Host Ports: 0/UDP, 0/TCP, 0/TCP
Args:
-conf
/etc/coredns/Corefile
State: Waiting
Reason: ContainerCreating
Ready: False
Restart Count: 0
Limits:
memory: 170Mi
Requests:
cpu: 100m
memory: 70Mi
Liveness: http-get http://:8080/health delay=60s timeout=5s period=10s #success=1 #failure=5
Readiness: http-get http://:8181/ready delay=0s timeout=1s period=10s #success=1 #failure=3
Environment:
Mounts:
/etc/coredns from config-volume (ro)
/var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-9rj9b (ro)
Conditions:
Type Status
Initialized True
Ready False
ContainersReady False
PodScheduled True
Volumes:
config-volume:
Type: ConfigMap (a volume populated by a ConfigMap)
Name: coredns
Optional: false
kube-api-access-9rj9b:
Type: Projected (a volume that contains injected data from multiple sources)
TokenExpirationSeconds: 3607
ConfigMapName: kube-root-ca.crt
ConfigMapOptional:
DownwardAPI: true
QoS Class: Burstable
Node-Selectors: kubernetes.io/os=linux
Tolerations: CriticalAddonsOnly op=Exists
node-role.kubernetes.io/control-plane:NoSchedule
node-role.kubernetes.io/master:NoSchedule
node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
Type Reason Age From Message

Normal Scheduled 3m53s default-scheduler Successfully assigned kube-system/coredns-64897985d-7lcxm to k8s-w-02
Warning FailedCreatePodSandBox 3m53s kubelet Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "105a61b6b61f191cb29d62f7e1a2bbd30e4caf737cc32ae0a4606e4f2900c6e4": unable to allocate IP address: Post "http://127.0.0.1:6784/ip/105a61b6b61f191cb29d62f7e1a2bbd30e4caf737cc32ae0a4606e4f2900c6e4": dial tcp 127.0.0.1:6784: connect: connection refused
Warning FailedCreatePodSandBox 3m39s kubelet Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "bf402a61c5b81398d955bc9394d55cdb7768355dbceb738f300307cbe8faf2e8": unable to allocate IP address: Post "http://127.0.0.1:6784/ip/bf402a61c5b81398d955bc9394d55cdb7768355dbceb738f300307cbe8faf2e8": dial tcp 127.0.0.1:6784: connect: connection refused
Warning FailedCreatePodSandBox 3m28s kubelet Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "ad0b1fca42a70750a2d4c6e507ecdcd93de3ce8a72485fefdb2f9c81bfd697c3": unable to allocate IP address: Post "http://127.0.0.1:6784/ip/ad0b1fca42a70750a2d4c6e507ecdcd93de3ce8a72485fefdb2f9c81bfd697c3": dial tcp 127.0.0.1:6784: connect: connection refused
Warning FailedCreatePodSandBox 3m14s kubelet Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "5bda39f04f1b90e188ec89d43cc748b00599041d15db8c4826826710e8a92fab": unable to allocate IP address: Post "http://127.0.0.1:6784/ip/5bda39f04f1b90e188ec89d43cc748b00599041d15db8c4826826710e8a92fab": dial tcp 127.0.0.1:6784: connect: connection refused
Warning FailedCreatePodSandBox 3m1s kubelet Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "0d512c5f83924b549c8aaeb853df5a9043a68f5eb351cdff299833b2300d1b22": unable to allocate IP address: Post "http://127.0.0.1:6784/ip/0d512c5f83924b549c8aaeb853df5a9043a68f5eb351cdff299833b2300d1b22": dial tcp 127.0.0.1:6784: connect: connection refused
Warning FailedCreatePodSandBox 2m49s kubelet Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "176fc507dba62986ae335e952b948414e3d47ebaff9d0b11544834509edf1705": unable to allocate IP address: Post "http://127.0.0.1:6784/ip/176fc507dba62986ae335e952b948414e3d47ebaff9d0b11544834509edf1705": dial tcp 127.0.0.1:6784: connect: connection refused
Warning FailedCreatePodSandBox 2m37s kubelet Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "482391f26c3c72b7a2a4e652ec8c987811e04c501f155b499a52e340f7e281cb": unable to allocate IP address: Post "http://127.0.0.1:6784/ip/482391f26c3c72b7a2a4e652ec8c987811e04c501f155b499a52e340f7e281cb": dial tcp 127.0.0.1:6784: connect: connection refused
Warning FailedCreatePodSandBox 2m26s kubelet Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "91460edd39da6a53650064566a821d9ce6920e62e191810868207aa9619310ed": unable to allocate IP address: Post "http://127.0.0.1:6784/ip/91460edd39da6a53650064566a821d9ce6920e62e191810868207aa9619310ed": dial tcp 127.0.0.1:6784: connect: connection refused
Warning FailedCreatePodSandBox 2m10s kubelet Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "232b239577c9c59bb9eb1f5d6f282d3530a6e3655fb3fecefd50b8a2c9a47fa4": unable to allocate IP address: Post "http://127.0.0.1:6784/ip/232b239577c9c59bb9eb1f5d6f282d3530a6e3655fb3fecefd50b8a2c9a47fa4": dial tcp 127.0.0.1:6784: connect: connection refused
Warning FailedCreatePodSandBox 13s (x9 over 116s) kubelet (combined from similar events): Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "9def87ec78c96208a2ed5075419a5c811df41b7ae4a55e3f8c892f0de4d7c83d": unable to allocate IP address: Post "http://127.0.0.1:6784/ip/9def87ec78c96208a2ed5075419a5c811df41b7ae4a55e3f8c892f0de4d7c83d": dial tcp 127.0.0.1:6784: connect: connection refused

I even commented on the Loop and reflected on the Pods, but the result was no different.

**Comment by chrisohaver** [comment]
> Hello, could anyone tell me if this would be a bug 

This is not a CoreDNS bug.

**Comment by chrisohaver** [comment]
Probably a misconfiguration with your K8s cluster CNI. Try searching "Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox".

**Reply by vitorafgomes** [reply]
A while ago I had used this setting to boot k8s and it was working.

sudo kubeadm init --control-plane-endpoint "k8s-haproxy" --upload-certs

I even tried the same configuration, but it gave me an error in coredns.

sudo kubeadm init

on both I'm using WAVE

kubectl apply -f "https://cloud.weave.works/k8s/net?k8s-version=$(kubectl version | base64 | tr -d '\n')"

but still, it doesn't work

**Reply by chrisohaver** [reply]
> I even tried the same configuration, but it gave me _an error in coredns_.

The error is not with CoreDNS. The errors are from kubelet, a core Kubernetes component.  The error is more or less that kubelet could not fully provision a Pod because it could not get an IP address for it from the CNI. There is no interaction with the CoreDNS image here, CoreDNS hasn't started yet at this point.

The error summarized ...

> Warning ... kubelet Failed to create pod sandbox: ... failed to setup network for sandbox ...  unable to allocate IP address: ... connection refused

My guess is that your CNI is not installed properly.

**Reply by vitorafgomes** [reply]
Thanks a lot for the help.

**Comment by vitorafgomes** [comment]
[chrisohaver](https://github.com/chrisohaver) my problem was the linux version
